File: leetcode/001-Two-Sum/scala/Solution.scala
object Solution {
  def twoSum(nums: Array[Int], target: Int): Array[Int] = {
    var numIdx: Map[Int, Int] = Map()
    for((num, i) <- nums.view.zipWithIndex) {
      if (numIdx.contains(target - num)) {
        return Array(numIdx.get(target - num).getOrElse(-1), i)
      }
      numIdx += (num -> i)
    }
    return Array[Int]()
  }
}

File: leetcode/001-Two-Sum/swift/Solution.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var numIdx = [Int: Int]()
        for (i, num) in nums.enumerated() {
            if numIdx[target - num] != nil {
                return [numIdx[target - num]!, i]
            }
            numIdx[num] = i
        }
        return []
    }
}